weblate/vcs/git.py

```diff
@@ -1468,3 +1468,125 @@ def create_pull_request(
 
         if "id" not in response:
             raise RepositoryException(0, f"Pull request failed: {error_message}")
+
+
+class BitbucketServerRepository(GitMergeRequestBase):
+
+    name = gettext_lazy("Bitbucket Server pull request")
+    identifier = "bitbucketserver"
+    _version = None
+    API_TEMPLATE = "https://{host}/rest/api/1.0/projects/{owner}/repos/{slug}"
+    bb_fork: Dict = {}
+
+    def get_headers(self, credentials: Dict):
+        headers = super().get_headers(credentials)
+        headers["Authorization"] = f"Bearer {credentials['token']}"
+        return headers
+
+    def create_fork(self, credentials: Dict):
+        fork_url, owner, slug = self.get_api_url()
+        bb_fork, error_message = self.request("post", credentials, fork_url, json={})
+        self.bb_fork = bb_fork
+
+        # If fork already exists, get forks from origin and find the user's fork.
+        # Since Bitbucket uses projectKey which can be any string(that is not
+        # at all related to its name) we need to compare user's forks against
+        # remote.
+        if "This repository URL is already taken." in error_message:
+            page = 0
+            self.bb_fork = {}
+            while True:
+                forks_url = f'{credentials["url"]}/forks?limit=1000&start={page}'
+                forks, error_message = self.request("get", credentials, forks_url)
```

Comment:
This could use params as well instead of generating the URL in each loop:

```suggestion
            forks_url = f'{credentials["url"]}/forks'
            while True:
                forks, error_message = self.request("get", credentials, forks_url, params={"limit": 1000, "start": page})
```

Reply:
fixed in amend.